tutorials/e2e-genai-langchain-app/src/backend/main.py

```diff
@@ -0,0 +1,32 @@
+from flask import Flask, request, jsonify
+from flask_cors import CORS
+from model import init_ray_and_deploy
+import requests
+import logging
+
+app = Flask(__name__)
+CORS(app)
+logging.basicConfig(level=logging.INFO)
+
+RAY_ENDPOINT = 'http://example-cluster-kuberay-head-svc:8000'  # Consider moving to configuration
+
+init_ray_and_deploy()
```

Comment:
If model with the same class name (`DeployLLM`) is already deployed, it redeploys it. Should we be incrementing the name and checking if the last version is already running? Is there a recommended practice? Not sure we need to introducing version logic or a separate model deployer service for this tutorial. What do you think?

Reply:
This raises a good question: what is the recommended approach for deploying ray serve applications? I think the answer is a [RayService](https://docs.ray.io/en/master/cluster/kubernetes/user-guides/rayservice.html#), but I'm interested in @richardsliu's perspective.

Reply:
I personally don't like the RayService CRD, primarily because the serve config is a yaml string nested in another yaml: https://docs.ray.io/en/master/cluster/kubernetes/user-guides/rayservice.html#step-3-install-a-rayservice

In general my recommendation would be to keep an autoscaling-enabled Ray cluster running while submitting Ray Serve jobs to it. Different versions of the model could be served under different routes.

For the purpose of this tutorial I'm fine with leaving it as it is.